users.js

```diff
@@ -0,0 +1,208 @@
+let database = `{
+    "users": [
+        {
+            "id": "1234567890",
+            "name": "mhitrov",
+            "email": "m.hitrov@gmail.com",
+            "age": 17,
+            "address": "Simeon Veliki street 62, Troyan, Bulgaria"
+        },
+        {
+            "id": "1234567891",
+            "name": "mkoparanov",
+            "email": "mkoparanov@gmail.com",
+            "age": 21,
+            "address": "Simeon Veliki street 60, Troyan, Bulgaria"
+        },
+        {
+            "id": "1234567892",
+            "name": "vvalkov",
+            "email": "vvalkov@gmail.com",
+            "age": 24,
+            "address": "Vezhen street 28, Troyan, Bulgaria"
+        },
+        {
+            "id": "1234567893",
+            "name": "yvalkova",
+            "email": "yvalkova@gmail.com",
+            "age": 19,
+            "address": "Makedonia street 21, Troyan, Bulgaria"
+        },
+        {
+            "id": "1234567894",
+            "name": "svelikov",
+            "email": "svelikov@gmail.com",
+            "age": 20,
+            "address": "u tqh"
+        }
+    ]
+}`;
+
+//Using this object as a reference to the table rows.
+let trMap = {};
+
+const nameRegex = /[A-za-z]{1,30}/;
+const emailRegex = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+const addressRegex = /\w{1,300}/;
+
+if (!localStorage.getItem("database")) {
+  localStorage.setItem("database", database);
+}
+
+let tableBody = document
+  .getElementById("usersTable")
+  .getElementsByTagName("tbody")[0];
+
+const getUsers = async () => {
+  let usersJSON = localStorage.getItem("database") || database;
+  return JSON.parse(usersJSON).users;
+};
+
+const addUserToTable = (user) => {
+  let tableRow = document.createElement("tr");
+
+  let idTd = document.createElement("td");
+  let nameTd = document.createElement("td");
+  let emailTd = document.createElement("td");
+  let ageTd = document.createElement("td");
+  let addressTd = document.createElement("td");
+  let editTd = document.createElement("td");
+  let deleteTd = document.createElement("td");
+
+  let editButton = document.createElement("button");
+  let deleteButton = document.createElement("button");
+
+  editButton.innerHTML = "Edit";
+  editButton.className = "button";
+  editButton.addEventListener("click", () => loadUserToForm(user));
```

Comment:
Can you think of another way to handle these `edit` and `delete` actions? Currently you are assigning two event listeners for each row. There are better ways to do this. Hint: search for `event delegation`.